### PR TITLE
move scale forbidden windows to hpa spec

### PR DESCRIPTION
**Pull request by yarntime**
move scale forbidden windows to hpa spec.
_It is useful in my use cases, but I'm not sure whether we should refactor it_

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34424)

### CODE REVIEW

File: pkg/apis/autoscaling/types.go [hunk 1]
```diff
@@ -74,6 +74,10 @@ type HorizontalPodAutoscalerSpec struct {
 	// target average CPU utilization (represented as a percentage of requested CPU) over all the pods;
 	// if not specified the default autoscaling policy will be used.
 	TargetCPUUtilizationPercentage *int32 `json:"targetCPUUtilizationPercentage,omitempty"`
+	// downscale forbidden window minutes, defaults to 5 Minutes
```
Comment (erictune) [inline pkg/apis/autoscaling/types.go]: I think we usually use seconds for times in APIs.  Minutes might be too coarse.

**Comment by erictune** [comment]
An owner for the controllers code needs to review this too.

@beeps 
@derekwaynecarr 
@mikedanese 
you are pkg/controller/OWNERS
who would like to review?

File: pkg/apis/autoscaling/types.go (hunk 1 above)
Comment (yarntime) [inline pkg/apis/autoscaling/types.go]: ok, I will change the unit to seconds later

**Comment by yarntime** [comment]
@bprashanth 
@derekwaynecarr 
@mikedanese 
would you please take a moment to review? thank you very much.

**Comment by mikedanese** [comment]
cc @kubernetes/autoscaling

**Comment by piosz** [comment]
cc @mwielgus @fgrzadkowski

**Comment by jszczepkowski** [comment]
This idea already appeared in #32408: scale-up/scale-down windows are internal details of the autoscaling algorithm, so, we don't want to expose it in the API.

**Comment by fgrzadkowski** [comment]
Please read the whole discussion in https://github.com/kubernetes/kubernetes/pull/32408. We really don't want to expose this on the API level for number of reasons. It has been discussed over sig-autoscaling and @DirectXMan12 plans to fix some of the underlying issues to make this a) shorter and b) configurable on the controller level (flag to controller manager).

**Comment by yarntime** [comment]
@fgrzadkowski, @jszczepkowski , thank you for your replay,  I will read #32408 to synchronize the information.

**Pull request closed without merging**